ipaserver/install/server/install.py

```diff
@@ -471,6 +471,15 @@ def install_check(installer):
 
     domain_name = domain_name.lower()
 
+    if options.setup_dns:
+        api.env.domain = domain_name
```

Comment:
Why is this assignment necessary?

Reply:
inside install_check domain was not found inside api. api.env.domain.
So provided domain_name explicitly, and tested this does not break things.

Reply:
I don't like this approach. It feels more like a dirty, fragile hack than a properly solution. Either the function shouldn't assume a fully configured api object or the ``api.env`` should be finalized.

Reply:
Thankyou @tiran I was also thinking the same.Will change, test and do the change.